### ISSUE TITLE
Finder reports error -43 when copying symlinked directories (Mountain Lion)

### ISSUE THREAD

**Issue by maltoe** [osxfuse/osxfuse#75]
Using sshfs or [bindfs](http://code.google.com/p/bindfs/) as client filesystem, I get 'error -43' when trying to copy a symlinked directory *into* the mount point (for example, the osxfuse source directory contains several symlinks after compilation). The symlinks have to be *relative* for this error to occur.

Command line (none of the options have any effect on the error it seems):

    bindfs -u malte -o allow_other -o iosize=1048576 -o auto_xattr -o nolocalcaches -o volname=Project -o direct_io -o local -o noappledouble --chmod-ignore --chgrp-ignore --chown-ignore dir1 dir2

Specs:
- Mac OS X 10.8.2
- OSXFUSE 2.5.4
- bindfs 1.11
- sshfs 2.4.1

Some speculations:
* This may be related to the various [Samba+Error -43](http://apple.stackexchange.com/questions/68374/how-can-you-avoid-error-43-when-copying-symlinked-folder-in-finder-with-a-samba) posts on the net.
* It may be a result of the order of copy operations issued by Finder. When Finder stumbles upon a relative symlink targeted at a directory or file living in the same directory as the symlink, it seems to depend on the names of these files and their alphabetical order whether the errors pops up or not.
* In conclusion, Finder seems to try to copy the targets of symlinks, too (which is fine), but has trouble finding the *original* target of the symlink.

I'm relatively new to Mac OS X and am already quite stunned about the lack of documentation and help on the Web, however, from what I read I understood that Finder copy errors (in relation to foreign FSs) are quite common and often not easily fixed. Still, I'd like to know whether you think it'd be possible to find some kind of workaround to fix Finder through OSXFUSE? For now, I changed bindfs's bindfs_readlink and bindfs_symlink to always point symlinks to ".", but that's not really a solution.

Thanks & best regards
Malte

**Comment by bfleischer** [comment]
I've run into similar issues with the loopback reference file system. It turned out the `XATTR_NOFOLLOW` flag was missing in one or two xattr function calls. For many syscalls there are two variants, one that follows symlinks and another that does not. Its important to always use the variant that does NOT follow symlinks. Anyways, this is something the bindfs team would need to look into should it still be an issue.
